lib/recipe/bloc/recipe_bloc.dart

```diff
@@ -94,21 +95,126 @@ class RecipeBloc extends Bloc<RecipeEvent, RecipeState> {
+    List<Map<String, dynamic>> probes =
+        List.from(recipes[changedRecipeIndex].probes);
+    probes.add(probe);
+
+    Recipe recipe =
+        recipes[changedRecipeIndex].copyWith(probes: probes, saved: false);
+
+    recipes.removeAt(changedRecipeIndex);
+
+    recipes.insert(changedRecipeIndex, recipe);
+    emit(state.copyWith(recipes: recipes));
+  }
+
+  void _onProbeChanged(ProbeChanged event, Emitter<RecipeState> emit) {
+    List<Recipe> recipes = List.from(state.recipes);
+    int changedRecipeIndex = recipes.indexOf(event.recipe);
+
+    List<Map<String, dynamic>> probes =
+        List.from(recipes[changedRecipeIndex].probes);
+    int changedProbeIndex = event.probeIndex;
+
+    String probeName = event.probeName;
+    Map<String, dynamic> probe = {
+      'probe': probeName,
+      'key': probes[changedProbeIndex]['key']
+    };
+
+    probes.removeAt(changedProbeIndex);
+    probes.insert(changedProbeIndex, probe);
+
+    Recipe recipe =
+        recipes[changedRecipeIndex].copyWith(probes: probes, saved: false);
+
+    recipes.removeAt(changedRecipeIndex);
+    recipes.insert(changedRecipeIndex, recipe);
+
+    emit(state.copyWith(recipes: recipes));
+  }
+
+  void _onProbeDeleted(ProbeDeleted event, Emitter<RecipeState> emit) {
+    List<Recipe> recipes = List.from(state.recipes);
+    int changedRecipeIndex = recipes.indexOf(event.recipe);
+
+    List<Map<String, dynamic>> probes =
+        List.from(recipes[changedRecipeIndex].probes);
+    int changedProbeIndex = probes.indexOf(event.probe);
+    probes.removeAt(changedProbeIndex);
+
+    Recipe recipe =
+        recipes[changedRecipeIndex].copyWith(probes: probes, saved: false);
+
+    recipes.removeAt(changedRecipeIndex);
+
+    recipes.insert(changedRecipeIndex, recipe);
+    emit(state.copyWith(recipes: recipes));
+  }
+
   void _onIngredientAdded(IngredientAdded event, Emitter<RecipeState> emit) {
     List<Recipe> recipes = List.from(state.recipes);
+    Recipe standardRecipe = event.recipe;
     int changedRecipeIndex = recipes.indexOf(event.recipe);
 
     final ingredient = Ingredient();
     List<Ingredient> ingredients =
         List.from(recipes[changedRecipeIndex].ingredients);
     ingredients.add(ingredient);
 
-    Recipe recipe = recipes[changedRecipeIndex]
-        .copyWith(ingredients: ingredients, saved: false);
+    Recipe recipe = (event.recipe.saved == true &&
+            event.recipe.recipeType == "Standard Recipe")
+        ? recipes[changedRecipeIndex].copyWith(
+            ingredients: ingredients,
+            saved: false,
+            recipeType: "Modified Recipe",
+            recipeNumber: const Uuid().v4())
+        : recipes[changedRecipeIndex]
+            .copyWith(ingredients: ingredients, saved: false);
 
     recipes.removeAt(changedRecipeIndex);
+
+    if (event.recipe.saved == true &&
+        event.recipe.recipeType == "Standard Recipe") {
+      recipes.insert(changedRecipeIndex, standardRecipe);
+    }
```

Comment:
What exactly does this do?

Reply:
If the recipe being edited is a standard one and is saved, this creates a copy of that standard recipe which is inserted into the list. The recipe which continues to edited is now the modified recipe. If the recipe being edited is non-standard/modified, proceed as normal with updating the ingredient list and changing the saved state to false.

Reply:
I see, thanks 💙!